### PR TITLE
BROOKLYN-4381 : Add support for AT+ORP interface

### PR DESCRIPTION
**Pull request by gbetous-swir**

### CODE REVIEW

File: clients/c/src/orpClient.c [hunk 1]
```diff
@@ -316,13 +356,32 @@ static void orp_Dispatch
     // nothing implemented yet
 }
 
+//--------------------------------------------------------------------------------------------------
+/**
+ * Deframe and decode AT packets
```
Comment (Nico26-sw) [inline clients/c/src/orpClient.c:361]: Not sure this function really decodes AT packets :smile: 
`orp_AtDumpFrame` ?

**Review by Nico26-sw — CHANGES_REQUESTED** [review]
Hi Guillaume, I have some remarks and questions
Thanks

**Review by fdur — COMMENTED** [review]
Minor comment
I'll check deeply later

File: clients/c/src/at.c [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    strncpy(dest,AT_PREFIX,destlen);
+    dst_idx = strlen(AT_PREFIX);
+    
+    // ORP packet type
+    dest[dst_idx++]=src[src_idx++];
+
+    // ORP data type (may be NULL)
+    if (0 != src[src_idx])
+    {
+        dest[dst_idx++]=src[src_idx];
```
[the remark this answers is not in the document]
Comment (imorrisonca) [reply clients/c/src/at.c:77]: Yes, please use white space between operators such as `=`, `+`, etc.

**Review by imorrisonca — CHANGES_REQUESTED** [review]
Hi @gbetous-swir 

The changes look quite good.  I just had some minor comments around formatting.

Thanks,
Ian

File: clients/c/src/at.c [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+ * AT commands implementation
+ *
+ */
+
+#include "at.h"
+#include "legato.h"
+#include <string.h>
+
+static char* AT_PREFIX="AT+ORP=\"";
+static char* AT_SUFFIX="\"\n";
```
[the remark this answers is not in the document]
Comment (gbetous-swir) [reply clients/c/src/at.c:41]: I renamed at_suffix and at_prefix to copy style on hdlc.c (e.g. crc_tabccitt_init )

**Review by Nico26-sw — APPROVED** [review]
Look good to me, thanks !

**Review by imorrisonca — APPROVED** [review]
Looks good - thanks

**Pull request merged**